### PR TITLE
Don't use Flex shortcuts when installing packages

### LINKED ISSUE #9693: Should we use full Composer package names or Symfony Flex shortcuts?
Related to https://github.com/symfony/symfony-docs/pull/9691 and other similar PRs we've received. Our idea was to always assume that the reader of 3.4+ docs was using Flex, so we use shortcuts everywhere. Should we keep doing that?

My vote is to keep using shortcuts, but let's ask to doc maintainers: @weaverryan @xabbuh @wouterj @HeahDude. Thanks!

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #9693. I've left some shortcuts on purpose:

* in `frontend/encore/installation.rst`: `composer require encore` because Webpack Encore is a bit special and the shortcut is the best way to install it.
* in `quick_tour/*.rst` because it shows the features provided by Symfony and shortcuts is one of those features. Readers will learn about them and later they'll decide if they want to use them or not.
* in `setup/flex.rst` because they are used on purpose to talk about Flex shortcuts.

### CODE REVIEW

File: doctrine.rst [hunk 1]
```diff
@@ -23,8 +23,8 @@ code:
 
 .. code-block:: terminal
 
-    composer require doctrine
-    composer require maker --dev
+    $ composer require doctrine/doctrine-bundle
```
Comment (xabbuh) [inline doctrine.rst]: By the way, why don't we use `orm` here instead? With this step the required `doctrine/orm` package would be missing leading to errors.

File: doctrine.rst [hunk 2]
```diff
@@ -479,7 +479,7 @@ for you automatically! First, install the bundle in case you don't have it:
 
 .. code-block:: terminal
 
-    $ composer require annotations
+    $ composer require doctrine/annotations
```
Comment (xabbuh) [inline doctrine.rst]: That's not the same, `annotations` is an alias for the SensioFrameworkExtraBundle.

File: doctrine/registration_form.rst [hunk 3]
```diff
@@ -14,7 +14,7 @@ First, make sure you have all the dependencies you need installed:
 
 .. code-block:: terminal
 
-    $ composer require doctrine form security validator
+    $ composer require doctrine/doctrine-bundle symfony/form symfony/security-bundle symfony/validator
```
Comment (xabbuh) [inline doctrine/registration_form.rst]: Same as above: use `orm` instead?

File: page_creation.rst [hunk 4]
```diff
@@ -195,15 +195,15 @@ To use the web debug toolbar, just install it:
 
 .. code-block:: terminal
 
-    $ composer require --dev profiler
+    $ composer require --dev symfony/web-profiler-bundle
```
Comment (xabbuh) [inline page_creation.rst]: should be `symfony/profiler-pack`

File: profiler.rst [hunk 5]
```diff
@@ -12,7 +12,7 @@ install the profiler before using it:
 
 .. code-block:: terminal
 
-    $ composer require profiler --dev
+    $ composer require symfony/web-profiler-bundle --dev
```
Comment (xabbuh) [inline profiler.rst]: `symfony/profiler-pack`

File: routing.rst [hunk 6]
```diff
@@ -24,7 +24,7 @@ First, install the annotations package:
 
 .. code-block:: terminal
 
-    $ composer require annotations
+    $ composer require doctrine/annotations
```
Comment (xabbuh) [inline routing.rst]: SensioFrameworkExtraBundle

File: security.rst [hunk 7]
```diff
@@ -136,7 +136,7 @@ You'll learn later how to deny access to certain URLs or controllers.
 
     .. code-block:: terminal
 
-        $ composer require profiler --dev
+        $ composer require symfony/web-profiler-bundle --dev
```
Comment (xabbuh) [inline security.rst]: `symfony/profiler-pack`

File: serializer.rst [hunk 8]
```diff
@@ -123,7 +123,7 @@ To use annotations, first install the annotations package:
 
 .. code-block:: terminal
 
-    $ composer require annotations
+    $ composer require doctrine/annotations
```
Comment (xabbuh) [inline serializer.rst]: SensioFrameworkExtraBundle

**Comment by javiereguiluz** [comment]
@xabbuh thanks for the review. I'm sorry I made so many mistakes. Hopefully they are fixed now!

File: _includes/_annotation_loader_tip.rst.inc [hunk 9]
```diff
@@ -1,7 +1,7 @@
 .. note::
 
     In order to use the annotation loader, you should have installed the
-    ``doctrine/annotations`` and ``doctrine/cache`` packages with Composer.
+    ``sensio/framework-extra-bundle`` and ``doctrine/cache`` packages with Composer.
```
Comment (xabbuh) [inline _includes/_annotation_loader_tip.rst.inc]: I am not sure where we include this file, but is it needed in that context to have the SensioFrameworkExtraBundle or are these cases where `doctrine/annotations` is actually fine?

Comment (javiereguiluz) [reply _includes/_annotation_loader_tip.rst.inc]: We include this file in:

* /components/routing.rst
* /components/serializer.rst
* /components/validator/resources.rst

In 2.7, 2.8 and 3.4 we showed this recommendation:

```rst
.. note::

    In order to use the annotation loader, you should have installed the
    ``doctrine/annotations`` and ``doctrine/cache`` packages with Composer.
```

So let's keep it unchanged.

**Review by xabbuh — APPROVED** [review]

**Pull request closed without merging**